suite/suite.go

```diff
@@ -105,9 +105,18 @@ func Run(t *testing.T, suite TestingSuite) {
 		}
 	}
 
-	if !testing.RunTests(func(_, _ string) (bool, error) { return true, nil },
-		tests) {
-		t.Fail()
+	// Go 1.7 and up - check if t has Run method.
+	// Putting T into interface so we can do runtime rather than compile time type assert for compat reasons
+	var tInterface interface{} = t
+	if subT, hasSubtests := tInterface.(subtestT); hasSubtests {
+		for _, test := range tests {
+			subT.Run(test.Name, test.F)
+		}
+	} else { // Go 1.6 and below
+		if !testing.RunTests(func(_, _ string) (bool, error) { return true, nil },
+			tests) {
+			t.Fail()
+		}
```

Comment:
We could actually declare `allTestsFilter` globally and reuse it in `suite_test.go` to make it more readable too.